pkg/github/tokensource.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package github
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/abcxyz/pkg/githubauth"
+)
+
+// KeyProvider provides a private key.
+type KeyProvider interface {
+	Key(ctx context.Context) ([]byte, error)
+}
+
+type AppTokenSource struct {
+	keyProvider KeyProvider
+	appID       string
+}
+
+func NewAppTokenSource(keyProvider KeyProvider, appID string) *AppTokenSource {
+	return &AppTokenSource{
+		keyProvider: keyProvider,
+		appID:       appID,
+	}
+}
+
+func (s *AppTokenSource) TokenForOrg(ctx context.Context, orgID int64) (string, error) {
```

Comment:
FR: consider caching the token for the org.

Reply:
I've added this an issue.